pkg/component/observability/monitoring/alertmanager/alertmanager.go

```diff
@@ -73,6 +73,11 @@ func (a *alertManager) alertManager(takeOverOldPV bool) *monitoringv1.Alertmanag
 		obj.Spec.AlertmanagerConfiguration = &monitoringv1.AlertmanagerConfiguration{Name: a.name()}
 	}
 
+	if a.values.Replicas > 1 {
+		obj.Spec.PodMetadata.Labels["networking.resources.gardener.cloud/to-alertmanager-operated-tcp-9094"] = v1beta1constants.LabelNetworkPolicyAllowed
+		obj.Spec.PodMetadata.Labels["networking.resources.gardener.cloud/to-alertmanager-operated-udp-9094"] = v1beta1constants.LabelNetworkPolicyAllowed
+	}
```

Comment:
Can we please add a comment about service `alertmanager-operated` being automatically created/managed by prometheus operator? Otherwise, this looks like even more magic than usual.